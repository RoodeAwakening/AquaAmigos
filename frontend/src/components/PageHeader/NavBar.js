import React from 'react';
import { NavLink } from 'react-router-dom';
import LogoutButton from '../auth/LogoutButton';
import styles from './NavBar.module.css';

const NavBar = () => {
  return (
    <nav className={styles.NavBar_container}>
      <ul>
        <li>
          <NavLink to="/" exact activeClassName="active">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink to="/login" exact activeClassName="active">
            Login
          </NavLink>
        </li>
        <li>
          <NavLink to="/sign-up" exact activeClassName="active">
            Sign Up
          </NavLink>
        </li>
        <li>
          <NavLink to="/users" exact activeClassName="active">
            Users
          </NavLink>
        </li>
        <li>
          <LogoutButton />
        </li>
      </ul>
    </nav>
  );
};

export default NavBar;
