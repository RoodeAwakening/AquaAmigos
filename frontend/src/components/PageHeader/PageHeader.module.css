.pageHeader_container{
  /* blue gradient from the center out */
  
}

.pageHeader_top{
  background-color: #fff;
  box-shadow: 0px 0px 10px rgba(0,0,0,0.1);
}

.pageHeader_middle{
/* linear gradient from the center to the right and left */
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding-top: 25px;
  padding-bottom: 25px;
  margin:  0;
  background-color: #fff;
  box-shadow: 0px 0px 10px rgba(0,0,0,0.1);
}

.pageHeader_middle_user_icon{
  position: absolute;
  right : 5%;
}

.pageHeader_middle_logo img{
  width: auto;
  height: 7em;
  
}

.pageHeader_nav{
  background-color: #fff;
  box-shadow: 0px 0px 10px rgba(0,0,0,0.1);
}

@media screen and (max-width: 768px){
  .pageHeader_middle{
    display: flex;
    justify-content: space-around;
    background-image: none;
  }

  .pageHeader_middle_user_icon{
    position: relative;
  }
}
  
