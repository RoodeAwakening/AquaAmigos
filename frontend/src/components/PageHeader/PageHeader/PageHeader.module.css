.pageHeader_container{
  /* blue gradient from the center out */
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;

  
}

.pageHeader_top{
  background-color: #fff;
  box-shadow: 0px 0px 10px rgba(0,0,0,0.1);
}


.pageHeader_middle_user_icon{
  
}

.pageHeader_middle_logo img{
  width: auto;
  height: 7em;
  
}

.pageHeader_nav{
  background-color: #fff;
}

@media screen and (max-width: 768px){
  .pageHeader_middle{
    display: flex;
    justify-content: space-around;
    background-image: none;
  }

  .pageHeader_middle_user_icon{
    position: relative;
  }
}
  
