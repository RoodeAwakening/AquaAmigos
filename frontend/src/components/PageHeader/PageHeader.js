import React from 'react';
import NavBar from '../NavBar/NavBar';
import styles from './PageHeader.module.css';
import logo2 from '../../images/logo2.png';

function PageHeader() {
  return (
    <div className={styles.pageHeader_container}>
      <div className={styles.pageHeader_top}>
        {/* this will hold a searchbar at some point */}
      </div>
      <div className={styles.pageHeader_middle}>
        <div className={styles.pageHeader_middle_logo}>
          <img src={logo2} alt="logo" />
        </div>
        <div className={styles.pageHeader_middle_user_icon}>
          {/* { i18next.t(msg_edit_profile) } */}
          <h2>user placeholder</h2>
        </div>
      </div>
      <div className={styles.pageHeader_nav}>
        <NavBar />
      </div>
    </div>


  );
}

export default PageHeader;
