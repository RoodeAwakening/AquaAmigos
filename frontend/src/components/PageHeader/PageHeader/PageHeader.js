import React from 'react';
import NavBar from '../NavBar/NavBar';
import styles from './PageHeader.module.css';
// import logo2 from '../../images/logo2.png';
import logo from '../../../images/logo.jpg';
import UserIconBlock from '../UserIconBlock/UserIconBlock';

function PageHeader(sessionUser) {
  return (
    <div className="toast-top-center">
      <div className={styles.pageHeader_container}>
        <div className={styles.pageHeader_middle_logo}>
          <img src={logo} alt="logo" />
        </div>
        <div className={styles.pageHeader_nav}>
          <NavBar {...sessionUser} />
        </div>
        <div className={styles.pageHeader_middle_user_icon}>
          {/* { i18next.t(msg_edit_profile) } */}
          <UserIconBlock {...sessionUser} />
        </div>
      </div>
    </div>
  );
}

export default PageHeader;
