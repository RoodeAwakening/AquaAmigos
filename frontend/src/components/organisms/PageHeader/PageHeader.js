import React, { useEffect, useState } from 'react';
import NavBar from '../../molecules/NavBar/NavBar';
import styles from './PageHeader.module.css';
// import logo2 from '../../images/logo2.png';
import logo from '../../../images/logo.jpg';
import UserIconBlock from '../../molecules/UserIconBlock/UserIconBlock';

function PageHeader(sessionUser) {
  const [isOpen, setIsOpen] = useState(false);

  const showMenu = () => {
    setIsOpen(!isOpen);
  };

  useEffect(() => {
    // update menu settings based on device size.
    if (window.innerWidth > 768) {
      const mobileMenuButton = document.getElementById('mobileButton');
      mobileMenuButton.style.display = 'none';
    }
    if (window.innerWidth < 768) {
      const pageHeaderContainer = document.getElementById('pageHeader_container');
      if (isOpen) {
        pageHeaderContainer.style.display = 'block';
      } else {
        pageHeaderContainer.style.display = 'none';
      }
    }
  }, [isOpen]);

  return (
    <div className="toast-top-center">
      <div className={styles.pageHeader_mobile}>
        <div>
          <img src={logo} alt="logo" className={styles.pageHeader_mobile__logo} />
        </div>
        <div>
          <button type="button" className={styles.btn_mobile} onClick={showMenu}>
            <i className="fas fa-bars" />
          </button>
        </div>
      </div>
      <div className={styles.pageHeader_container} id="pageHeader_container">
        <div className={styles.pageHeader_middle_logo}>
          <img src={logo} alt="logo" />
        </div>
        <div className={styles.pageHeader_mobile__button} id="mobileButton">
          <button type="button" className={styles.btn_mobile} id="btn_mobile" onClick={showMenu}>
            <i className="fas fa-bars" />
          </button>
        </div>
        <div className={styles.pageHeader_nav}>
          <NavBar
            {...sessionUser}
            showMenu={showMenu}
          />
        </div>
        <div className={styles.pageHeader_middle_user_icon}>
          {/* { i18next.t(msg_edit_profile) } */}
          <UserIconBlock {...sessionUser} />
        </div>
      </div>
    </div>
  );
}

export default PageHeader;
