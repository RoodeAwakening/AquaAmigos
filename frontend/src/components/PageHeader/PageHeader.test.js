import React from 'react';
import { shallow, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import PageHeader from './PageHeader';

configure({ adapter: new Adapter() });
global.confirm = () => true;

function setup() {
  return shallow(<PageHeader />);
}

describe(' Page Header Snapshot', () => {
  let pageHeader;

  beforeEach(() => {
    pageHeader = setup();
  });

  afterEach(() => {
    pageHeader.unmount();
  });

  it(' Should render the page header and should match snapshot', () => {
    expect(toJson(PageHeader)).toMatchSnapshot();
  });
});
