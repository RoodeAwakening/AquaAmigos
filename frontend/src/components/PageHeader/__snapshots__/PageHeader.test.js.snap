// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Page Header Snapshot  Should render the page header and should match snapshot 1`] = `undefined`;
